pkg/key/common.go

```diff
@@ -15,3 +26,43 @@ func ControlPlane(getter LabelsGetter) string {
 func MachineDeployment(getter LabelsGetter) string {
 	return getter.GetLabels()[label.MachineDeployment]
 }
+
+// MaxBatchSizeIsValid will validate the value into valid maxBatchSize
+// valid values can be either:
+// an integer between 0 < x <= worker count
+// a float between 0 < x <= 1
+// float value is used as ratio of a total worker count
+func MaxBatchSizeIsValid(value string, maxWorkers int) bool {
```

Comment:
I put the validation function into the `key` package because I will have the exact same validation on AWSCluster and it felt weird to have exact same code in 2 places in one repo

Reply:
scratch that moved to aws package